### DISCUSSION TITLE
How to add a specific device id to the driver

### DISCUSSION THREAD

**Discussion by arutar** [NVIDIA/open-gpu-kernel-modules#426] [Q&A]
I want to add a specific PCI Device ID, PCI Subsystem Vendor ID, and PCI Subsystem Device ID to the driver. 
How can i do this?

**Comment by aritger** [comment]
In general, the open-gpu-kernel-modules don't (or at least shouldn't) use PCI device IDs for very much: once the driver sees the device is NVIDIA's vendor ID, it inspects the GPU architecture to wire up various HALs.

There are a few specialized tables of PCI IDs for things like whether to default to using GSP offload with the closed-source driver.

It may help to start with what problem you are trying to solve.  Do you have a particular NVIDIA device that is not working today?  What is the error when it doesn't work?  What is its PCI device ID?
